language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -240,17 +252,49 @@ void error(String message, Throwable t) {
 
     InitializeResult result = new InitializeResult();
     ServerCapabilities c = new ServerCapabilities();
-    TextDocumentSyncOptions textDocumentSyncOptions = new TextDocumentSyncOptions();
-    textDocumentSyncOptions.setOpenClose(true);
-    textDocumentSyncOptions.setChange(TextDocumentSyncKind.Full);
-    textDocumentSyncOptions.setSave(new SaveOptions(true));
-    c.setTextDocumentSync(textDocumentSyncOptions);
+    c.setTextDocumentSync(getTextDocumentSyncOptions());
     c.setCodeActionProvider(true);
     c.setExecuteCommandProvider(new ExecuteCommandOptions(asList(SONARLINT_OPEN_RULE_DESCRIPTION_COMMAND)));
+    c.setWorkspace(getWorkspaceServerCapabilities());
+
     result.setCapabilities(c);
     return CompletableFuture.completedFuture(result);
   }
 
+  private List<URI> toUriList(@Nullable List<WorkspaceFolder> workspaceFolders) {
+    List<URI> list = new ArrayList<>();
+    if (workspaceFolders == null) {
+      return list;
+    }
+
+    for (WorkspaceFolder folder : workspaceFolders) {
+      try {
+        list.add(new URI(folder.getUri()));
+      } catch (URISyntaxException e) {
+        debug("Could not parse URI from " + folder.getUri());
+      }
+    }
+    return list;
+  }
+
+  private WorkspaceServerCapabilities getWorkspaceServerCapabilities() {
+    WorkspaceFoldersOptions options = new WorkspaceFoldersOptions();
+    options.setSupported(true);
+    options.setChangeNotifications(true);
+
+    WorkspaceServerCapabilities capabilities = new WorkspaceServerCapabilities();
+    capabilities.setWorkspaceFolders(options);
+    return capabilities;
+  }
+
+  private TextDocumentSyncOptions getTextDocumentSyncOptions() {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "getTextDocumentSyncOptions" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.core%3Asonarlint-core-parent&pullRequest=191&issues=AWMHyHIHuCK54mAOyiXr&open=AWMHyHIHuCK54mAOyiXr)

[ISSUE_KEY]: AWMHyHIHuCK54mAOyiXr